README.md

```diff
@@ -1,213 +1,97 @@
+import { z } from 'zod';
 
-import { implementSchema } from '@lifeomic/one-schema';
-
-import { Schema } from './generated-api';
-
-const router = new Router();
-
-implementSchema(Schema, {
-  on: router,
-  implementation: {
-    'POST /items': (ctx) => {
-      // `ctx.request.body` is well-typed and has been run-time validated.
-      console.log(ctx.request.body.label);
-
-      // TypeScript enforces that this matches the `Response` schema.
-      return { id: '123', label: 'test label' };
-    },
-    'GET /items': (ctx) => {
-      // `ctx.request.query` is well-typed and has been run-time validated
-      console.log(ctx.request.query.filter);
-
-      // TypeScript enforces that this matches the `Response` schema.
-      return [{ id: '123', label: 'test label' }];
-    },
-  },
+const router = OneSchemaRouter.create({
+  using: new Router(),
   introspection: {
     route: '/private/introspection',
-    serviceVersion: process.env.LIFEOMIC_BUILD_ID!,
+    serviceVersion: process.env.LIFEOMIC_SERVICE_VERSION,
   },
-});
+})
+  .declare({
+    route: 'POST /items',
+    name: 'createItem',
+    request: z.object({ message: z.string() }),
+    response: z.object({ id: z.string(), message: z.string() }),
+  })
+  .declare({
+    route: 'GET /items/:id',
+    name: 'getItemById',
+    request: z.object({ filter: z.string() }),
+    response: z.object({ id: z.string(), message: z.string() }),
+  });
+
+// Be sure to expose your router's routes on a Koa app.
+import Koa from 'koa';
+
+const app = new Koa().use(router.middleware());
 
-const server = new Koa()
-  .use(router.routes())
-  .use(router.allowedMethods())
-  .listen();
+app.listen();
 ```
 
-By default, `implementSchema` will perform input validation on all of your routes, using the defined `Request` schemas.
-To customize this input validation, specify a `parse` function:
+Once you have routes declared, add implementations for each route. Enjoy perfect type inference and auto-complete for path parameters, query parameters, and the request body.
 
 ```typescript
-implementSchema(Schema, {
-  // ...
-  parse: (ctx, { endpoint, schema, data }) => {
-    // Validate `data` against the `schema`.
-    // If the data is valid, return it, otherwise throw.
-  },
+router
+  .implement('POST /items', async (ctx) => {
+    ctx.request.body; // { message: string }
+    return { id: 'some-id', message: ctx.request.body.message };
+  })
+  .implement('GET /items/:id', async (ctx) => {
+    ctx.request.query; // { filter: string }
+    ctx.params; // { id: string }
+    return { id: 'some-id', message:'some-id' };
+  })
 });
 ```
 
-### Axios Client Generation
+### Generating Type-Safe Clients
 
-Projects that want to safely consume a service that uses `one-schema` can perform introspection using `fetch-remote-schema`.
+To generate a type-safe client for this new API, we need to:
 
-```
+1. Introspect the deployed schema using the `one-schema` CLI. Commit this file.
```

Comment:
Should we list the required CLI commands here to be really obvious??

Reply:
I lay out the specific commands using code blocks just below.

Reply:
🤦🏻‍♂️